timestamp/http.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package timestamp
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+// maxBodyLength specifies the max content can be received from the possibly
+// malicious remote server.
+// The legnth of a regular TSA response with certificates is usually less than
+// 10 KiB.
+const maxBodyLength = 1 * 1024 * 1024 // 1 MiB
+
+// TimestampQuery is the content-type of timestamp query.
+// RFC 3161 3.4
+const TimestampQuery = "application/timestamp-query"
+
+// TimestampReply is the content-type of timestamp reply
+// RFC 3161 3.4
+const TimestampReply = "application/timestamp-reply"
+
+// httpTimestamper is a HTTP-based timestamper.
+type httpTimestamper struct {
+	rt       http.RoundTripper
+	endpoint string
+}
+
+// NewHTTPTimestamper creates a HTTP-based timestamper with the endpoint
+// provided by the TSA.
+// http.DefaultTransport is used if nil RoundTripper is passed.
+func NewHTTPTimestamper(rt http.RoundTripper, endpoint string) (Timestamper, error) {
+	if rt == nil {
+		rt = http.DefaultTransport
+	}
+	if _, err := url.Parse(endpoint); err != nil {
```

Comment:
Do we need to limit the request to be `https://`?

Reply:
Many of the free TSA servers provide the `http` endpoints, for example, the one for `digicert` is: `http://timestamp.digicert.com`.